test/integration/model.js

```diff
@@ -517,6 +517,33 @@ module.exports = function(bookshelf) {
         });
       });
 
+      it('ensure events are async', function(ok) {
```

Comment:
You call the events "async", but they seem more sync now since the next listener is only called after the first one resolves, so can you explain the async nature of this test?

Reply:
will update 👍